test/e2e/pv_reclaimpolicy.go

```diff
@@ -0,0 +1,231 @@
+			Expect(err).NotTo(HaveOccurred())
+
+			volumeoptions.CapacityKB = 2097152
+			volumeoptions.Name = "e2e-vmdk-" + strconv.FormatInt(time.Now().UnixNano(), 10)
+			volumeoptions.DiskFormat = "thin"
+			volumePath, err = vsp.CreateVolume(&volumeoptions)
+
+			Expect(err).NotTo(HaveOccurred())
+			pv = getVSpherePersistentVolumeSpec(volumePath, v1.PersistentVolumeReclaimDelete, nil)
+			pv, err := c.CoreV1().PersistentVolumes().Create(pv)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating the pv")
+			pvc = getVSpherePersistentVolumeClaimSpec(ns, nil)
+
+			pvc, err := c.CoreV1().PersistentVolumeClaims(ns).Create(pvc)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("wait for the pv and pvc")
+			waitOnPVandPVC(c, ns, pv, pvc)
+
+			By("delete pvc")
+			deletePersistentVolumeClaim(c, pvc.Name, ns)
+
+			By("verify pv is deleted")
+			err = framework.WaitForPersistentVolumeDeleted(c, pv.Name, 3*time.Second, 300*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	/*
+		This test Verify persistent volume should be retained when reclaimPolicy on the PV is set to retain
+		and associated claim is deleted
+
+		Test Steps:
+		1. Create vmdk
+		2. Create PV Spec with volume path set to VMDK file created in Step-1, and PersistentVolumeReclaimPolicy is set to Retain
+		3. Create PVC with the storage request set to PV's storage capacity.
+		4. Wait for PV and PVC to bound.
+		5. Delete PVC
+		6. Verify PV is retained.
+		7. Delete PV.
+	*/
+
+	framework.KubeDescribe("persistentvolumereclaim:retain", func() {
+		var (
+			volumePath    string
+			pv            *v1.PersistentVolume
+			pvc           *v1.PersistentVolumeClaim
+			volumeoptions vsphere.VolumeOptions
+		)
+
+		It("should retain persistent volume when reclaimPolicy set to retain when associated claim is deleted", func() {
+
+			By("creating vmdk")
+			vsp, err := vsphere.GetVSphere()
+			Expect(err).NotTo(HaveOccurred())
+
+			volumeoptions.CapacityKB = 2097152
+			volumeoptions.Name = "e2e-vmdk-" + strconv.FormatInt(time.Now().UnixNano(), 10)
+			volumeoptions.DiskFormat = "thin"
+			volumePath, err = vsp.CreateVolume(&volumeoptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating the pv")
+			pv = getVSpherePersistentVolumeSpec(volumePath, v1.PersistentVolumeReclaimRetain, nil)
+			pv, err := c.CoreV1().PersistentVolumes().Create(pv)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating the pvc")
+			pvc = getVSpherePersistentVolumeClaimSpec(ns, nil)
+
+			pvc, err := c.CoreV1().PersistentVolumeClaims(ns).Create(pvc)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("wait for the pv and pvc")
+			waitOnPVandPVC(c, ns, pv, pvc)
+
+			By("delete pvc")
+			deletePersistentVolumeClaim(c, pvc.Name, ns)
+
+			By("verify pv is retained")
+			framework.Logf("Waiting for PV %v to become Released", pv.Name)
+			err = framework.WaitForPersistentVolumePhase(v1.VolumeReleased, c, pv.Name, 3*time.Second, 300*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
```

Comment:
Do you want to add a code here to verify the data on the volume has been preserved?